src/pages/patientView/mutationalSignatures/MutationalSignaturesContainer.tsx

```diff
@@ -51,50 +126,247 @@ export default class MutationalSignaturesContainer extends React.Component<
+                        'btn-outline-secondary': this.selectedScale !== scale,
+                    },
+                    'btn',
+                    'btn-sm',
+                    'btn-axis-switch'
+                )}
+                data-test={`AxisScaleSwitch${scale}`}
+                style={{
+                    lineHeight: 1,
+                    cursor:
+                        this.selectedScale === scale ? 'default' : 'pointer',
+                    fontWeight:
+                        this.selectedScale === scale ? 'bolder' : 'normal',
+                    color: this.selectedScale === scale ? '#fff' : '#6c757d',
+                    backgroundColor:
+                        this.selectedScale === scale ? '#6c757d' : '#fff',
+                    borderColor: '#6c757d',
+                }}
+                onClick={onClick}
+            >
+                {scale}
+            </button>
+        );
+    }
+    @observable yAxisLabel: string = 'Mutational count (value)';
+
+    @action.bound
+    private handlePercentClick() {
+        this.selectedScale = AxisScale.PERCENT;
+        this.yAxisLabel = 'Mutational count (%)';
+    }
+
+    @action.bound
+    private handleCountClick() {
+        this.selectedScale = AxisScale.COUNT;
+        this.yAxisLabel = 'Mutational count (value)';
+    }
+
+    @computed get getDataForGraph(): IMutationalCounts[] {
+        if (this.selectedScale == AxisScale.PERCENT) {
+            return getPercentageOfMutationalCount(
+                this.mutationalSignatureCountDataGroupedByVersionForSample
+            );
+        } else {
+            return this.mutationalSignatureCountDataGroupedByVersionForSample;
+        }
+    }
+
+    @autobind
+    private getSvg() {
+        if (document.getElementById('mutationalBarChart')) {
+            return document.getElementById('mutationalBarChart')!
+                .firstChild as SVGElement;
+        } else {
+            return null;
+        }
+    }
+
+    @observable
+    _mutationalSignatureCountDataGroupedByVersionForSample: IMutationalCounts[];
+    @computed get mutationalSignatureCountDataGroupedByVersionForSample() {
+        const sampleIdToFilter = this.props.samples.includes(this.props.sample)
+            ? this.props.sample
+            : undefined;
+        return (
+            this._mutationalSignatureCountDataGroupedByVersionForSample ||
+            this.props.dataCount[this.props.version]
+                .map(item => item)
+                .filter(subItem => subItem.sampleId === sampleIdToFilter)
+        );
+    }
+
+    @action.bound changeSignature(name: string): void {
+        this._selectedSignature = name;
+    }
+
     @action.bound
     private onVersionChange(option: { label: string; value: string }): void {
         this.props.onVersionChange(option.value);
+        this.signatureProfile = this.props.data[option.value][0].meta.name;
+        this.signatureInformationToolTipVisible = false;
+    }
+
+    @action.bound
```

Comment:
doesn't really need to be action because it isn't directly mutating state.  it's the callback (props.onSampleChange) which probably needs to be action.